File: kanito/Classi/CustomControls/MyLabel/MyLabel.m

//  MyLabel.m
//  csen cinofilia
//
//  Created by Luciano Calderano on 25/03/16.
//  Copyright © 2016 Kanito. All rights reserved.
//

#import "MyLabel.h"

@implementation MyLabel

- (instancetype)init {
    self = [super init];
    if (self)
        [self defaultMyLabel];
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self)
        [self defaultMyLabel];
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self)
        [self defaultMyLabel];
    return self;
}

- (void) defaultMyLabel {
    self.minimumScaleFactor = 0.75;
    self.adjustsFontSizeToFitWidth = YES;
    self.font = [MyFont fontSize:self.font.pointSize Type:MyFontTypeSemiBold];
    self.title = self.text;
//    if ([[self.text substringToIndex:1] isEqualToString:@"#"])
//        self.text = keyLang([self.text substringFromIndex:1]);
    //    self.textColor = defCol_Blu;
}

- (void) setTitle:(NSString *)title {
    if (title.length > 0 && [[title substringToIndex:1] isEqualToString:@"#"]) {
        self.text = keyLang([title substringFromIndex:1]);
    }
    else {
        self.text = title;
    }
}
@end

File: Swift/Cls/CustomClass/MYLabel.swift
//
//  MYButton.swift
//  Enci
//
//  Created by Luciano Calderano on 03/11/16.
//  Copyright © 2016 Kanito. All rights reserved.
//

import UIKit

class MYLabel: UILabel {
    var title: String {
        get {
            return self.text!
        }
        set {
            self.text = newValue.tryLang()
        }
    }
    required internal init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.initialize()
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.initialize()
    }
    override internal func awakeFromNib() {
        super.awakeFromNib()
        self.initialize()
    }
    
    private func initialize () {
        self.minimumScaleFactor = 0.75
        self.adjustsFontSizeToFitWidth = true
        self.font = UIFont.mySize(self.font.pointSize)
        if (self.text != nil) {
            self.title = self.text!
        }
    }
}
